src/main/c/output.c

```diff
@@ -605,3 +605,73 @@ jobject output_convert_schema_checked(JNIEnv *env, const ddwaf_object *obj)
     JNI(DeleteLocalRef, ret);
     return NULL;
 }
+
+jobject convert_ddwaf_object_to_jobject(JNIEnv *env, const ddwaf_object *obj)
+{
+    switch (obj->type) {
+        // TODO: Implement missing types, when needed
+        case DDWAF_OBJ_INVALID:
+        case DDWAF_OBJ_SIGNED:
+        case DDWAF_OBJ_UNSIGNED:
+        case DDWAF_OBJ_FLOAT:
+        case DDWAF_OBJ_BOOL:
+            return NULL;
+        case DDWAF_OBJ_STRING:
+            return java_utf8_to_jstring_checked(env, obj->stringValue, obj->nbEntries);
+        case DDWAF_OBJ_ARRAY: {
+            jobject ret = java_meth_call(env, &_array_list_init, NULL, (jint) obj->nbEntries);
```

Comment:
it would be better to check if `obj->nbEntries > INT_MAX` before the cast and if yes throw an exception.

Because this is called only on the actions returned it's not strictly required now though

Reply:
Good point, fixed